src/test/java/com/linkedin/kmf/services/ConsumeServiceTest.java

```diff
@@ -0,0 +1,186 @@
+
+    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
+    List<MetricsReporter> reporters = new ArrayList<>();
+    reporters.add(new JmxReporter(Service.JMX_PREFIX));
+    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
+    String name = "name";
+    Map<String, String> tags = new HashMap<>();
+    tags.put(TAGS_NAME, name);
+
+    _commitAvailabilityMetrics = new CommitAvailabilityMetrics(metrics, tags);
+    Assert.assertNotNull(_commitAvailabilityMetrics._offsetsCommitted.name());
+    Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+    Assert.assertEquals(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue(), 0.0);
+
+    /* Should start */
+    consumeService.start();
+    Assert.assertTrue(consumeService.isRunning());
+
+    long threadStartDelay = 2000;
+
+    /* Thread.sleep safe to do here instead of ScheduledExecutorService
+    *  We want to sleep current thread so that consumeService can start running for enough seconds. */
+    Thread.sleep(threadStartDelay);
+    Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+    Assert.assertTrue(testCommittedSuccessfully);
+
+    consumeService.stop();
+    consumeService.stop();
+
+    consumeService.awaitShutdown();
+  }
+
+  /**
+   * Sample ConsumeService instance for unit testing
+   * @return Sample ConsumeService object.
+   * @throws Exception
+   */
+  private ConsumeService consumeService() throws Exception {
+    LOG.info("Creating an instance of Consume Service for testing..");
+
+    ConsumerFactory consumerFactory = Mockito.mock(ConsumerFactory.class);
+    AdminClient adminClient = Mockito.mock(AdminClient.class);
+    KMBaseConsumer kmBaseConsumer = Mockito.mock(KMBaseConsumer.class);
+
+    Mockito.when(consumerFactory.adminClient()).thenReturn(adminClient);
+    Mockito.when(consumerFactory.latencySlaMs()).thenReturn(20000);
+    Mockito.when(consumerFactory.baseConsumer()).thenReturn(kmBaseConsumer);
+    Mockito.when(consumerFactory.topic()).thenReturn(TOPIC);
+
+    /* LATENCY_PERCENTILE_MAX_MS_CONFIG, */
+    Mockito.when(consumerFactory.latencyPercentileMaxMs()).thenReturn(5000);
+
+    /* LATENCY_PERCENTILE_GRANULARITY_MS_CONFIG */
+    Mockito.when(consumerFactory.latencyPercentileGranularityMs()).thenReturn(1);
+
+    /* define return value */
+    long mockLastCommittedOffset = 0;
+    Mockito.when(kmBaseConsumer.lastCommitted()).thenReturn(mockLastCommittedOffset);
+    Mockito.doAnswer(new Answer() {
+      @Override
+      public String answer(InvocationOnMock invocation) {
+        testCommittedSuccessfully = true;
+        return "Mock Offset Committed Successfully.";
+      }
+    }).when(kmBaseConsumer).commitAsync(Mockito.any());
```

Comment:
The callback supplied to commitAsync needs to be called inside this method.

This looks relevant.

https://www.baeldung.com/mockito-callbacks

Reply:
Done.